lib/ansible/modules/files/ini_file.py

```diff
@@ -203,11 +211,12 @@ def do_ini(module, filename, section=None, option=None, value=None,
                     # change the existing option line
                     if match_opt(option, line):
                         newline = assignment_format % (option, value)
-                        changed = ini_lines[index] != newline
-                        if changed:
+                        option_changed = ini_lines[index] != newline
```

Comment:
Previously, `changed` could only be true if the option was changed. Now, it can already be `True` if we have added an ending newline to the file. This is why I have modified this.